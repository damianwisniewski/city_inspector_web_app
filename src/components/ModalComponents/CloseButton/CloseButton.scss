@import '../../../assets/styles/colors';

.close-button {
	position: absolute;
	top: -15px;
	right: -15px;
	border-radius: 50%;
	width: 37px;
	height: 37px;
	background-color: transparent;
	z-index: 600;
	display: flex;
	justify-content: center;
	align-items: center;
	border: 0;
	outline: 0;
	cursor: pointer;

	&::before {
		padding: 0 !important;
		margin: 0;
		color: aliceblue;
		font-size: 26px !important;
		border-radius: 50%;
		border: 3px solid $primary_dark_blue;
		background-color: $primary_dark_blue;
		box-shadow: -1px 1px 3px 0 rgb(65, 65, 65);
	}
}
