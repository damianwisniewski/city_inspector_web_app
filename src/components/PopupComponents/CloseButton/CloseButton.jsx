import './CloseButton.scss'
import icons from '../../../assets/styleModules/icons.module.scss'
import React from 'react'
import PropTypes from 'prop-types'

const CloseButton = ({ onClose }) => (
	<button id='popupCloseButton' onClick={onClose} className={`close-button ${icons.exit}`} />
)

CloseButton.propTypes = {
	onClose: PropTypes.func.isRequired,
}

export default CloseButton
