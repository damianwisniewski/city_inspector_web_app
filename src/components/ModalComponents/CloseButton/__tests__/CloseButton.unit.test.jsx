import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import CloseButton from '../CloseButton'

configure({ adapter: new Adapter() })

describe('(Modal) <CloseButton />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let CloseButtonComponent
	const onClose = jest.fn()

	beforeAll(() => {
		CloseButtonComponent = shallow(<CloseButton onClose={onClose} />)
	})

	/**
	 * Snapshot
	 */
	test('(Non-Auth) should contains proper Navigation items', () => {
		expect(toJson(CloseButtonComponent)).toMatchSnapshot()
	})

	test('should call "onClose" method from props, onClick event', () => {
		CloseButtonComponent.simulate('click')
		expect(onClose).toBeCalledTimes(1)
	})
})
