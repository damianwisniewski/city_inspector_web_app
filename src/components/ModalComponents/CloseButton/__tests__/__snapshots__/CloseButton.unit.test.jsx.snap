// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Modal) <CloseButton /> (Non-Auth) should contains proper Navigation items 1`] = `
<button
  className="close-button exit"
  id="ModalCloseButton"
  onClick={[MockFunction]}
/>
`;
